tools/cross/format.py

```diff
@@ -102,6 +104,25 @@ def prettier(files: List[str], check: bool = False) -> bool:
     return result.returncode == 0
 
 
+def ruff(files: List[str], check: bool = False) -> bool:
+    if files and not shutil.which(RUFF):
+        msg = "Cannot find ruff, will not format Python"
+        if check:
+            # In ci, fail.
+            logging.error(msg)
+            return False
+        else:
+            # In a local checkout, let it go. If the user wants Python
+            # formatting they can install ruff and run again.
+            logging.warning(msg)
+            return True
+    cmd = [RUFF, "format"]
```

Comment:
Do we have any kind of "verbose" flag for ruff so we see what files it formats or checks?

Reply:
Actually, I was just about to open a pull-request to remove it, since it reduces formatting performance. Any particular reason to enable it?

Reply:
I find it convenient for local use to know which files were checked, and if they needed their formatting corrected or not.

Reply:
Agree with @npaun also, formatting shouldn't take too long for regular runs where a user just has a few files staged for commits.
The CI run might take a while but it's gonna be faster than build&test anyway.

Reply:
@anonrig  What kind of time difference did you see between with verbose vs without?

Reply:
I've opened a pull-request about 2 changes that I think will fulfill everybody's needs. https://github.com/cloudflare/workerd/pull/2576

> @anonrig What kind of time difference did you see between with verbose vs without?

I'm not sure. I have to benchmark it using a tool like `hyperfine` to came up with a scientific result.

Reply:
Yea sounds right.

Reply:
> I find it convenient for local use to know which files were checked, and if they needed their formatting corrected or not.

Unfortunately I don't think that ruff has a mode that does this without printing way too much output.